http_server/main_test.go

```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestWelcome(t *testing.T) {
+
+	req := httptest.NewRequest("GET", "localhost:8080/welcome?name=mara", nil)
```

Comment:
nit: You can omit `localhost:8080` in this case. `example.com` will be used, but that's alright. Otherwise it suggests that the test needs to somehow run on `:8080`.